src/index.ts

```diff
@@ -1,41 +1,45 @@
-import * as React from 'react';
-
 import optimized from './optimized';
 import { mapLoadable } from './bundler';
 import { Component, EnhancedPreLoadable, PreLoadable } from './interface';
-import { getComponent, isCached } from './map';
+import { cache, getComponent } from './map';
 
 const defaultPreLoadable: EnhancedPreLoadable = {
-    cached: true,
     placeholder: null,
     extract: 'default',
 };
 
 let i = 0;
 
+export const isCached = cache.has;
+
 const register = <P extends {}>(component: PreLoadable & Partial<EnhancedPreLoadable>) => {
+    const name = `Component${i++}`;
     const enhancedComponent: Component = {
-        name: `Component${i++}`,
+        name,
         ...defaultPreLoadable,
         ...component
     };
-    const { name } = enhancedComponent;
 
-    if (mapLoadable[name] !== undefined) {
+    if (mapLoadable.get(name) !== undefined) {
         console.warn(`You are trying to add a new component with already existing name: ${name}. If you see this warning after fast/hot reloading and you use 'register' function in Navigator without separate 'index' files - it's fine. In other cases it may indicate an issue, since by default it will overwrite the data and screen will not be cached. See https://github.com/kirillzyusko/react-native-bundle-splitter/issues/29 for more details.`);
     }
 
-    mapLoadable[name] = enhancedComponent;
+    mapLoadable.set(name, enhancedComponent);
 
     return optimized<P>(name);
 };
 
 const component = (name: string) => getComponent(name);
 
-const group = (name: string) => {
-    const components = Object.keys(mapLoadable).filter((componentName) => mapLoadable[componentName].group === name);
+const group = (groupName: string) => {
+    const components: ReturnType<(typeof getComponent)>[] = [];
+    mapLoadable.forEach(({ group, name }) => {
```

Comment:
Overall, why do we need to change the approach with pushing the data into array? Can't we rewrite it to something like:

```ts
const component = mapLoadable.keys().filter((key) => mapLoadable.get(key).group === groupName)
```

Reply:
because the keys method returns an iterator instead of an array 🙂 
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map/keys